### DISCUSSION TITLE
How --default-language option works?

### DISCUSSION THREAD

**Discussion by alefbragin** [dandavison/delta#858] [Q&A] [answered]
Some times `delta` cannot infers language from a filename. And it is pretty obvious that `--default-language` targets this cases:

```
        --default-language <default-language>
            Default language used for syntax highlighting when this cannot be inferred from a filename. It will
            typically make sense to set this in per-repository git config (.git/config)
```

But I try to utilize it in several ways, and neither is successful.

```bash
git config delta.default-language 'Bourne Again Shell (bash)'
git diff
```

```bash
git -c delta.default-language='Bourne Again Shell (bash)' diff
```

```bash
delta --default-language 'Bourne Again Shell (bash)' file1 file2
```

Moreover there are open feature requests with similar purposes: #485, #499. Do I correctly understand purpose of the option? Or maybe the option is buggy?

**Comment by dandavison** [comment, marked as answer]
Hi @alefbragin, I'm sorry, you hit upon an option which I had introduced for one purpose, but not announced, and was not doing what one would expect. I've fixed that now in https://github.com/dandavison/delta/pull/859 (so if you can compile Rust and would like to test that branch then great!). Here's what I wrote there:


The `default-language` option had been introduced initially in order to determine language for `git blame` (where there are no file names). However, subsequently `git blame` came to determine the language by inspecting the command line of the calling process (e.g. if delta sees the calling process is `git blame src/file.ts` then the language is typescript). Meanwhile `default-language` never actually acquired its obvious function of providing a default language for diffs; this PR fixes that.

Note that the best place to set `default-language` is probably in your per-repository git settings. I.e. `$my_repo/.git/config`.

**Reply by alefbragin** [reply]
Hi @dandavison! Thanks for the great utility and for the fix! I build that branch and it works as expected in my cases.
I think the fix helps @zachriggle in the [case](https://github.com/dandavison/delta/issues/485#issuecomment-878686614).
And this commit possibly close #499 because this is exactly what was requested, but author of request needs it for bash's process substitution (`<(...)`) that doesn't work with delta.

**Reply by eitau** [reply]
@dandavison can you specify that currently `default-language` option only applies to `blame` and, as far I can tell, `grep` subcommands in [the cli's `help` output](https://github.com/dandavison/delta/blob/master/src/cli.rs#L315)? I just got quite frustrated trying to debug why it isn't working in general case for past hour. Cheers

**Comment by dandavison** [comment]
The argument to `default-language` is a file extension, without the `.` So for example, this works on that branch:

```
git -c delta.default-language=bash diff
```

or

```
git -c delta.default-language=sh diff
```

Or, add to per-repository git config (`$myrepo/.git/config`):

```gitconfig
[delta]
    default-language = bash # or "sh"
```
